chromium_src/chrome/browser/ui/views/tabs/tab_strip_layout_helper.cc

```diff
@@ -16,25 +18,38 @@ std::vector<gfx::Rect> CalculateVerticalTabBounds(
 
   std::vector<gfx::Rect> bounds;
   gfx::Rect rect;
-  rect.set_width(width.value_or(tabs.front().GetPreferredWidth()));
-  rect.set_height(layout_constants.tab_height);
   for (const auto& tab : tabs) {
+    rect.set_x(tab.is_tab_in_group()
+                   ? BraveTabGroupHeader::GetLeftPaddingForVerticalTabs()
+                   : 0);
+    rect.set_width(width.value_or(tabs.front().GetPreferredWidth()) - rect.x());
+    rect.set_height(
+        tab.state_.open() == TabOpen::kOpen ? layout_constants.tab_height : 0);
     bounds.push_back(rect);
-    // Workaround to check if a tab is closing now. There's a tight coupling
-    // with TabWidthConstraints::TransformForPinnednessAndOpeneness()'s
-    // implementation.
-    const bool is_opened = tab.GetPreferredWidth() != TabStyle::GetTabOverlap();
-    if (is_opened)
+    if (tab.state_.open() == TabOpen::kOpen)
       rect.set_y(rect.bottom());
   }
   return bounds;
 }
 
-#define CalculateTabBounds                                                    \
-  use_vertical_tabs_ ? CalculateVerticalTabBounds(layout_constants,           \
-                                                  tab_widths, tabstrip_width) \
-                     : CalculateTabBounds
+#define FillGroupInfo                                 \
```

Comment:
Wow, this is magic :laughing:

Reply:
Thanks. I'm flattered 🤭

Reply:
> Wow, this is magic 😆

came here to say this :) but upon further inspection I haven't found `FillGroupInfo` in the upstream file. In fact this macro is used right on the next line, which means it can be replaced with a static function.

Why we'd better to replace this with a function: our rebase team has a helper script which ensures that we don't have defines in `chromium_src` files that are not mapped to something that will be replaced in upstream files, so this macro will most likely trigger this warning.
https://github.com/brave/brave-core/blob/57a6b133ef11745bf1256a4b27e94f441448887f/script/check_chromium_src.py#L128-L132

Reply:
> this macro is used right on the next line, which means it can be replaced with a static function.

Good point! we can still use short-circuiting with static function.

> ur rebase team has a helper script which ensures that we don't have defines in chromium_src files that are not mapped to something that will be replaced in upstream files,

Wow, that's really cool stuff. I'm always surprised by how many you've automated things. Thanks for letting me know that.

Reply:
> I'm always surprised by how many you've automated things.

no-no, not me 😁  that's the rebase team! huge kudos to them 👍